rasa_nlu_examples/classifiers/sparse_naive_bayes_intent_classifier.py

```diff
@@ -47,215 +26,14 @@ def required_components(cls) -> List[Type[Component]]:
         "class_prior": None,
     }
 
-    def __init__(
-        self,
-        component_config: Optional[Dict[Text, Any]] = None,
-        clf: Optional["sklearn.naive_bayes.BernoulliNB"] = None,
-        le: Optional["sklearn.preprocessing.LabelEncoder"] = None,
-    ) -> None:
-        """Construct a new naive Bayes intent classifier using the sklearn framework."""
-        from sklearn.preprocessing import LabelEncoder
-
-        super().__init__(component_config)
-
-        if le is not None:
-            self.le = le
-        else:
-            self.le = LabelEncoder()
-        self.clf = clf
-
-    @classmethod
-    def required_packages(cls) -> List[Text]:
-        return ["sklearn"]
-
-    def transform_labels_str2num(self, labels: List[Text]) -> np.ndarray:
-        """
-        Transforms a list of strings into numeric label representation.
-
-        :param labels: List of labels to convert to numeric representation
-        :returns: numpy array of numeric label ids.
-        """
-
-        return self.le.fit_transform(labels)
-
-    def transform_labels_num2str(self, y: np.ndarray) -> np.ndarray:
-        """
-        Transforms a numpy array of numeric label ids into a list of string label ids.
+    def create_sklearn_classifier(
+        self, **kwargs: Any
+    ) -> "sklearn.base.ClassifierMixin":
```

Comment:
Out of curiosity, why are you referring to the `sklearn.base.ClassifierMixin` as a string?

Reply:
because of the lazy imports